app/src/androidTest/java/org/mozilla/focus/activity/robots/SearchRobot.kt

```diff
@@ -57,6 +60,22 @@ class SearchRobot {
         assertTrue(searchBar.text.equals(text))
     }
 
+    fun verifySearchEditBarIsEmpty() {
+        searchBar.waitForExists(waitingTime)
+        assertTrue(searchBar.text.equals(getStringResource(R.string.urlbar_hint)))
+    }
+
+    fun verifyKeyboardVisibility(isExpectedToBeVisible: Boolean) {
+        mDevice.waitForIdle(waitingTime)
+
+        assertEquals(
+            isExpectedToBeVisible,
+            mDevice
+                .executeShellCommand("dumpsys input_method | grep mInputShown")
+                .contains("mInputShown=true"),
+        )
+    }
```

Comment:
Can we use the `InputMethodManager` and maybe `isAcceptingText()`? I don't think we need to execute shell commands. As well, I would treat this as a helper method outside of this robot.

https://developer.android.com/reference/android/view/inputmethod/InputMethodManager

Reply:
I was just porting this from Fenix.

I could give it a try.

Reply:
Thanks for the suggestion @AaronMT , made the changes and successfully re-ran the test 50x on Firebase.